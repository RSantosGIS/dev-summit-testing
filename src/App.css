.contentRoot {
  height: 93%;
  width: 100%;
  /*Good enough for now to fix the sizing problem with map and the nav bar
  TODO: Need a way to fix the stupid map dissapearing if not 100% of container problem
  Maybe add something to resize this to viewportHeight-64 px on load and window resize?
  */
}
.mapDiv {
  padding: 0px;
  height: 100%;
  width: 100%;
}

.esri-legend {
  max-height: 250px !important;
}
