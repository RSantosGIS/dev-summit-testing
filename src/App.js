import React from "react";
import Grid from '@material-ui/core/Grid';
import { BrowserRouter} from 'react-router-dom';
import AppBanner from "./components/AppBanner/AppBanner";
import ContentContainer from "./components/ContentContainer/ContentContainer";
import config from 'AgcShowcaseConfig';

//CSS
import "./App.css"; 

/**
 * The React Application root.  Uses Browserrouter for browser history and navigation routing
 * @author santosrj
 */
function App() {
  
  const tabIndex = config.tabDefs.tabIndexes;

  return <>
  <BrowserRouter basename={config.rootPath}>
    <Grid container>
      <Grid item xs={12}>
        <AppBanner tabIndex={tabIndex} />
      </Grid>
    </Grid>
    <Grid className="contentRoot"  container>
      <ContentContainer tabIndex={tabIndex}/>
    </Grid>
  </BrowserRouter>
  </>;
}

export default App;
