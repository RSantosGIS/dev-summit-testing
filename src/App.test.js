/* eslint-disable testing-library/no-node-access */
import { render, screen, fireEvent } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import App from './App';

const shallowRenderer = new ShallowRenderer();

//COMPONENT UNIT TESTS
describe ("App component unit tests", () => {
  test('renders App div', () => {
    shallowRenderer.render(<App />); //shallow render of just app component
    const view = shallowRenderer.getRenderOutput();
    expect(view.props.className).toEqual('App');
  });
});

//COMPONENT INTEGRATION AND SNAPSHOT TESTS
describe ("App wide integration and snapshot tests", () => {

  test ('ArcGIS integration connectivity', async () => {
    const response = await fetch('https://www.arcgis.com/sharing/rest/content/items/b6c889ff1f684cd7a65301984b80b93d?f=json', {
      headers: {
        'Content-Type': 'application/json'
      },
    });
    const jsonBody = await response.json();
    expect(jsonBody.id).toEqual('b6c889ff1f684cd7a65301984b80b93d');
    expect(jsonBody.type).toEqual('Web Scene');
  });

  it ('Matches snapshot', () => {
    const htmlTree = renderer.create(<App/>).toJSON();
    expect(htmlTree).toMatchSnapshot();
  });

  test ('End to End app integration and functionality test', () => {
    render(<App/>); //full render of entire React component hierarchy
    let areaBtn = screen.getByRole('button', {name: /AreaButton/i});
    
    //click distance Button
    let classNames = areaBtn.className.split(' ');
    expect(classNames).not.toContainEqual('active');
    fireEvent.click(areaBtn);
    classNames = areaBtn.className.split(' ');
    expect(classNames).toContainEqual('active');
  });
});

