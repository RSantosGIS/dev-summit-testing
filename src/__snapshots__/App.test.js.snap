// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App wide integration and snapshot tests Matches snapshot 1`] = `
<div
  className="App"
  data-testid="AppModule"
>
  <div
    className="sceneDiv"
    id="sceneDiv"
  />
  <div
    className="topbar"
  >
    <button
      aria-label="DistanceButton"
      className="action-button esri-icon-measure-line"
      title="Measure distance between two points"
      type="button"
    />
    <button
      aria-label="AreaButton"
      className="action-button esri-icon-measure-area"
      title="Measure Area"
      type="button"
    />
  </div>
</div>
`;
